Adds third-party secrets guide (#13173)
PrefectHQ/prefect: docs/guides/index.md

| [Webhooks](/guides/webhooks/) | Receive, observe, and react to events from other systems. |
| [Terraform Provider](https://registry.terraform.io/providers/PrefectHQ/prefect/latest/docs/guides/getting-started) | Use the Terraform Provider for Prefect Cloud for infrastructure as code. |
| [CI/CD](/guides/ci-cd/) | Use CI/CD with Prefect. |
| [Specifying Upstream Dependencies](/guides/specifying-upstream-dependencies/) | Run tasks in a desired order. |

> | [Third-party Secrets](/guides/secrets/) | Connect to services without storing credentials in blocks. |

Reviewer: ```suggestion
| [Third-party Secrets](/guides/secrets/) | Use credentials stored in a secrets manager in your workflows. |
```

Reviewer: Someone looking at this list may not know what a block is.
